srcCpp03/RTIDDSImpl.cxx

```diff
@@ -564,6 +589,110 @@ class RTIPublisher: public RTIPublisherBase<T> {
     }
 };
 
+#ifdef RTI_FLATDATA_AVAILABLE
+  template<typename T>
+  class RTIFlatDataPublisher: public RTIPublisherBase<T> {
+  protected:
+      typedef typename rti::flat::flat_type_traits<T>::builder Builder;
+      typedef typename rti::flat::PrimitiveArrayOffset<unsigned char, 4> KeyBuilder;
+      typedef typename rti::flat::PrimitiveSequenceBuilder<unsigned char> BinDataBuilder;
+
+      void add_key(Builder &builder, unsigned long int i) {
+          KeyBuilder key_offset = builder.add_key();
+
+          for (int j = 0; j < KEY_SIZE; ++j) {
+              // The key will be i but splitted in bytes
+              key_offset.set_element(j, (unsigned char) (i >> j * 8));
+          }
+      }
+
+  public:
+      RTIFlatDataPublisher(
+              dds::pub::DataWriter<T> writer,
+              int num_instances,
+              rti::core::Semaphore& pongSemaphore,
+              bool useSemaphore,
+              int instancesToBeWritten,
+              ParameterManager *PM
+              )
```

Comment:
coding conventions: the placement of `)` and `:` is incorrect:

```
RTIFlatDataPublisher(
    ...)
    : RTIPublisher...
```

Reply:
Corrected